src/geometry/ArborX_Ray.hpp

```diff
@@ -112,7 +113,30 @@ bool intersection(Ray const &ray, Box<3> const &box, float &tmin, float &tmax)
   {
     float tdmin;
     float tdmax;
-    if (dir[d] >= 0)
+    if (dir[d] == 0.)
```

Comment:
```suggestion
    if (dir[d] == 0)
```
Dunno if it matters. Just want to make sure `double` floating units are not used.